library/alloc/src/vec.rs

```diff
@@ -2928,6 +2972,37 @@ impl<T> IntoIter<T> {
         self.ptr = self.buf.as_ptr();
         self.end = self.buf.as_ptr();
     }
+
+    /// Shifts the remaining elements to `offset` and then converts the whole allocation into a Vec
+    /// with `vec.len() == offset + self.len()`
+    ///
+    /// # Safety
+    ///
+    /// When a non-zero offset is passed the resulting Vec will have an uninitialized prefix
+    /// that needs to be filled before the Vec is valid again.
+    ///
+    /// * `offset + self.len()` must not exceed `self.cap`
+    /// * `offset == 0` is always valid
+    /// * `offset` must be positive
+    unsafe fn into_vec_with_uninit_prefix(self, offset: isize) -> Vec<T> {
+        debug_assert!(offset > 0);
```

Comment:
This debug_assert is actually wrong I hope this doesn't blow up the perf run 😅

Reply:
I'm pretty sure perf builds without debug-assertions enabled, so you should be ok. This is a larger problem: the debug assertions in the standard library ~basically never run, because it's always distributed without assertions via rustup.

Reply:
I had an [auto merge build blow up](https://github.com/rust-lang/rust/pull/70793#issuecomment-686043351) on a debug assert in a previous PR. So they do something and are of some use, they just get tested pretty late in the PR process.